packages/pages/src/upgrade/pagesUpdater.ts

```diff
@@ -132,19 +139,34 @@ function processDirectoryRecursively(
  * @param targetDirectory
  */
 export const updateToUsePagesComponents = async (targetDirectory: string) => {
-  await updatePackageDependency(
+  const hasSitesComponents = await removePackageDependency(
     targetDirectory,
-    "@yext/pages-components",
-    null
+    "@yext/sites-components"
+  );
+  const hasReactComponents = await removePackageDependency(
+    targetDirectory,
+    "@yext/react-components"
   );
-  await removePackageDependency(targetDirectory, "@yext/sites-components");
-  await removePackageDependency(targetDirectory, "@yext/react-components");
   // update imports from pages/components to sites-components
-  replacePagesSlashComponentsImports(targetDirectory);
+  const hasPagesSlashComponentsImports =
+    replacePagesSlashComponentsImports(targetDirectory);
   // update imports from sites-components to pages-components
-  replaceSitesComponentsImports(targetDirectory);
+  const hasSitesComponentsImports =
+    replaceSitesComponentsImports(targetDirectory);
   // update imports from react-components to pages-components
-  replaceReactComponentsImports(targetDirectory);
+  const hasReactComponentsImports =
+    replaceReactComponentsImports(targetDirectory);
+
+  await updatePackageDependency(
+    targetDirectory,
+    "@yext/pages-components",
+    null,
+    hasSitesComponents ||
```

Comment:
We actually only want to add pages-components if one has been replaced, not if the previous packages were there too. Like if they were installed but not used, no point in adding an new unused package.